src/components/CampsiteListItem.js

```diff
@@ -22,64 +24,97 @@ const renderIcons = ({features, facilities}) => {
         )
     });
 
-    const facilityIcons = _.map(facilities, facility =>{
-       return(
-         <Icon
-             key={facility}
-             name={facilityIconDetails[facility].name}
-             type={facilityIconDetails[facility].type}
-         />
-       );
+    const facilityIcons = _.map(facilities, facility => {
+        return (
+            <Icon
+                key={facility}
+                name={facilityIconDetails[facility].name}
+                type={facilityIconDetails[facility].type}
+            />
+        );
     });
 
     return _.concat(featureIcons, facilityIcons);
 };
 
-const CampsiteListItem = ({site, getSiteDetail, navigate}) => {
-    const {textStyle, IconContainer, subtitleView, titleContainerStyle, siteAvatarContainerStyle, titleView, nearestTownStyle, badgeWrapperStyle, badgeContainerStyle, accessibilityStyle} = styles
-    const {campsite_form: {accessibility_options}} = campsite;
-    const {id, title, description, nearestTown, accessibility, siteImageData, features, facilities} = site;
-    const preparedDescription = description && description.length > 57 ?  `${description.substring(0, 57)}...` : description;
-
-    return (
-        <ListItem
-            bottomDivider={true}
-            onPress={() => getSiteDetail({selectedSite: site, navigate})}
-            leftAvatar={{avatarStyle: siteAvatarContainerStyle, source: siteImageData ? {uri: `data:image/png;base64,${siteImageData}`} : require('../../assets/starTent.jpg')}}
-            title={
-                <View style={titleContainerStyle}>
-                    <Text style={titleView}>{title}</Text>
-                    <Badge
-                        wrapperStyle={badgeWrapperStyle}
-                        containerStyle={badgeContainerStyle}
-                    >
-                        <Text style={nearestTownStyle}>{nearestTown}</Text>
-                    </Badge>
-                    <View style={subtitleView}>
-                        <Text>{preparedDescription}</Text>
-                    </View>
-                    <View style={subtitleView}>
+class CampsiteListItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            siteImageData: require(`../../assets/defaultSiteImage.jpg`)
+        }
+    }
 
-                        <Text style={accessibilityStyle}>{accessibility_options[accessibility]}</Text>
+    replaceImageData = () => {
+        returnImageForSiteKey({siteKey: this.props.site.id})
+            .then(imageData => {
+                this.setState({
+                    siteImageData: {uri: `data:image/png;base64,${imageData}`}
+                })
+            });
+    };
```

Comment:
This pattern is used in all places that show `siteImageData`. We use some local component-level state to show a placeholder image, and then once the real `siteImageData` comes back from `AsyncStorage` we replace it and call `setState`.